Packs/HYASInsight/Integrations/HYASInsight/HYASInsight.py

```diff
@@ -0,0 +1,838 @@
+        readable_output=whois_current_lookup_to_markdown(whois_current_record, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{WHOIS_CURRENT_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=api_response,
+        # outputs=[whois_current_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_malware_samples_records_by_indicator(client, args):
+    indicator_type = args.get('indicator_type')
+    indicator_value = args.get('indicator_value')
+
+    check_valid_indicator_type(indicator_type, MALWARE_QUERY_PARAMS)
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(MALWARE_SUB_CONTEXT, indicator_type, indicator_value)
+    end_point = MALWARE_ENDPOINT
+
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, False, 'POST')
+
+    return CommandResults(
+        readable_output=malware_samples_lookup_to_markdown(api_response, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{MALWARE_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=api_response,
+        # outputs=[malware_samples_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_associated_ips_by_hash(client, args):
+    indicator_type = MD5_PARAM
+    indicator_value = args.get('md5')
+
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(HASH_IP_SUB_CONTEXT, indicator_type, indicator_value)
+    end_point = MALWARE_ENDPOINT
+
+    associated_ips = []
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, False, 'POST')
+    if api_response:
+        for obj in api_response:
+            if obj['ipv4']:
+                associated_ips.append({'ip': str(obj['ipv4'])})
+
+    return CommandResults(
+        readable_output=associated_ips_lookup_to_markdown(associated_ips, title),
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.{HASH_IP_SUB_CONTEXT}',
+        outputs_key_field='',
+        outputs=associated_ips,
+        # outputs=[malware_samples_build_result_context(r) for r in new_response],
+    )
+
+
+@logger
+def get_associated_domains_by_hash(client, args):
+    indicator_type = MD5_PARAM
+    indicator_value = args.get('md5')
+
+    check_valid_indicator_value(indicator_type, indicator_value)
+    title = get_command_title_string(HASH_DOMAIN_SUB_CONTEXT, indicator_type, indicator_value)
+
+    end_point = MALWARE_ENDPOINT
+    if not re.match(md5Regex, indicator_value):
+        raise Exception(
+            f'Invalid indicator_value: {indicator_value} for indicator_type {indicator_type}')
+
+    associated_domains = []
+    api_response = client.fetch_data_from_hyas_api(end_point, indicator_type, indicator_value, False, 'POST')
+    if api_response:
+        for obj in api_response:
+            if obj['domain']:
+                associated_domains.append({'domain': str(obj['domain'])})
```

Comment:
Same as said above, please change outputs in YML accordingly.
```suggestion
    associated_domains = [str(obj['domain']) for obj in api_response if 'domain' in obj]
    outputs = {'md5': indicator_value, 'domains': associated_domains}
```